projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/memory/LLVMMemory.java

```diff
@@ -490,6 +501,14 @@ public void putVector(LLVMAddress address, LLVMAddressVector vector) {
         }
     }
 
+    public void putVector(LLVMAddress address, LLVMFunctionVector vector) {
+        long currentPtr = address.getVal();
+        for (int i = 0; i < vector.getLength(); i++) {
+            putAddress(currentPtr, vector.getAddress(i));
```

Comment:
See other comment in `getAddress`. Instead, use a `LLVMToNativeNode`. You can pass the node in as argument to the function, and create it using `@Cached("createToNativeWithTarget()")` at the callsite.